### ISSUE TITLE
TypeScript import/order regression

### ISSUE THREAD

**Issue by joaovieira** [benmosher/eslint-plugin-import#1525]
I've just upgraded to `eslint-plugin-import@2.18.2` and `eslint-import-resolver-typescript@2.0.0` and started to get `import/order` errors:

<img width="686" alt="Screen Shot 2019-10-29 at 17 28 05" src="https://user-images.githubusercontent.com/474603/67812811-f6416f80-fa97-11e9-96a8-98059a96457f.png">

Any module defined in `@types/*` (e.g. `@types/express` in this case) is seen as "internal", while modules with their own types (e.g. `http-status-codes`) are seen as "external". I've tried with other modules as well.

Changing the default order to:

```
    'import/order': [
      'error',
      {
        groups: ['internal', 'external'],
      },
    ],
```

Gets rid of the error.

Can't understand the root cause though, but here's some logging from the resolver if it helps:
<img width="1101" alt="Screen Shot 2019-10-29 at 22 09 36" src="https://user-images.githubusercontent.com/474603/67813201-d9596c00-fa98-11e9-9002-37b2fce46b4d.png">

The [`import/order` rule history](https://github.com/tihonove/eslint-plugin-import/commits/master/src/rules/order.js) doesn't seem to have anything suspicious. Could the new version of the resolver be causing this?

Re: https://github.com/alexgorbatchev/eslint-import-resolver-typescript/issues/33

**Comment by joaovieira** [comment]
And this is the resolver log with `eslint-import-resolver-typescript`:

<img width="1044" alt="Screen Shot 2019-10-29 at 22 39 34" src="https://user-images.githubusercontent.com/474603/67814853-23445100-fa9d-11e9-8896-67e87d9d975f.png">

As you see, the new version resolves the correct definition under `node_modules/@types/**/*.d.ts`, while the old one resolved the original `node_modules/**/*.js`.

**Comment by ljharb** [comment]
Yes, that sounds like it's due to the typescript resolver.

**Comment by joaovieira** [comment]
And the last evidence 🙂:

<img width="1108" alt="Screen Shot 2019-10-29 at 22 54 33" src="https://user-images.githubusercontent.com/474603/67815572-33f5c680-fa9f-11e9-9aa8-15b83c1a1e13.png">

<img width="1049" alt="Screen Shot 2019-10-29 at 22 52 14" src="https://user-images.githubusercontent.com/474603/67815563-293b3180-fa9f-11e9-939b-ea7b13193748.png">

Will look at the code now.

**Comment by joaovieira** [comment]
So, the package name is `express`, path is `node_modules/@types/express/index.d.ts`:
* it passes the external regexp
* but it's not external because `path.indexOf('node_modules/express') == -1`

We could simply use the `import/external-module-folders` setting to include the `@types` folder as `['node_modules', 'node_modules/@types']` in the typescript config, or change the `isExternalPath` logic to account for these scenarios.

@ljharb what do you think is best? I'm worried that with the setting it doesn't solve the problem for people that are already overriding it.

**Comment by Evalon** [comment]
Answer from eslint-import-resolver-typescript:
 https://github.com/alexgorbatchev/eslint-import-resolver-typescript/issues/34

**Comment by South-Paw** [comment]
I've also suddenly got this issue in our projects at work.

@Evalon's linked issue has been closed and locked and now @ljharb has closed this issue.

So what's actually the suggested solution here? Are we waiting on something to be merged or does something else need doing somewhere?

**Comment by ljharb** [comment]
@South-Paw in fact, it was closed by the PR #1526, which has been released - do you not see that in the timeline entry?

**Comment by South-Paw** [comment]
@ljharb sorry I totally missed that 🤦‍♂

I think the issue I'm seeing is unrelated to this one though, so sorry for digging up the wrong issue to reply to 😟 - I've created #1643
